packages/tangle/booker.go

```diff
@@ -119,30 +163,72 @@ func (m *Booker) allTransactionsApprovedByMessage(transactionIDs ledgerstate.Tra
 // indirectly approved by the given Message.
-func (m *Booker) transactionApprovedByMessage(transactionID ledgerstate.TransactionID, messageID MessageID) (approved bool) {
-	for _, attachmentMessageID := range m.tangle.Storage.AttachmentMessageIDs(transactionID) {
-		if m.tangle.Utils.MessageApprovedBy(attachmentMessageID, messageID) {
+func (b *Booker) transactionApprovedByMessage(transactionID ledgerstate.TransactionID, messageID MessageID) (approved bool) {
+	for _, attachmentMessageID := range b.tangle.Storage.AttachmentMessageIDs(transactionID) {
+		if b.tangle.Utils.MessageApprovedByStrongParents(attachmentMessageID, messageID) {
 			return true
 		}
 	}
 
 	return false
 }
 
-// branchIDsOfStrongParents returns the BranchIDs of the strong parents of the given Message.
-func (m *Booker) branchIDsOfStrongParents(message *Message) (branchIDs ledgerstate.BranchIDs) {
+// branchIDsOfParents returns the BranchIDs of the parents of the given Message.
+func (b *Booker) branchIDsOfParents(message *Message) (branchIDs ledgerstate.BranchIDs) {
 	branchIDs = make(ledgerstate.BranchIDs)
+
 	message.ForEachStrongParent(func(parentMessageID MessageID) {
-		if parentMessageID != EmptyMessageID && !m.tangle.Storage.MessageMetadata(parentMessageID).Consume(func(messageMetadata *MessageMetadata) {
+		if !b.tangle.Storage.MessageMetadata(parentMessageID).Consume(func(messageMetadata *MessageMetadata) {
 			branchIDs[messageMetadata.BranchID()] = types.Void
 		}) {
 			panic(fmt.Errorf("failed to load MessageMetadata with %s", parentMessageID))
 		}
 	})
 
+	message.ForEachWeakParent(func(parentMessageID MessageID) {
+		if parentMessageID == EmptyMessageID {
+			return
+		}
+		if !b.tangle.Storage.Message(parentMessageID).Consume(func(message *Message) {
+			if payload := message.Payload(); payload != nil && payload.Type() == ledgerstate.TransactionType {
+				transactionID := payload.(*ledgerstate.Transaction).ID()
+
+				if !b.tangle.LedgerState.utxoDAG.TransactionMetadata(transactionID).Consume(func(transactionMetadata *ledgerstate.TransactionMetadata) {
+					branchIDs[transactionMetadata.BranchID()] = types.Void
+				}) {
+					panic(fmt.Errorf("failed to load TransactionMetadata with %s", transactionID))
+				}
+			}
+
+		}) {
+			panic(fmt.Errorf("failed to load MessageMetadata with %s", parentMessageID))
+		}
+	})
+
 	return
 }
 
+func (b *Booker) createGenesis() {
+	genesisMetadata := &MessageMetadata{
+		messageID: EmptyMessageID,
+		solid:     true,
+		branchID:  ledgerstate.MasterBranchID,
+		structureDetails: &markers.StructureDetails{
+			Rank:          0,
+			IsPastMarker:  false,
+			PastMarkers:   markers.NewMarkers(),
+			FutureMarkers: markers.NewMarkers(),
+		},
+		timestampOpinion: TimestampOpinion{
+			Value: opinion.Like,
+			LoK:   Three,
+		},
+		booked:   true,
+		eligible: true,
+	}
+	b.tangle.Storage.messageMetadataStorage.Store(genesisMetadata).Release()
```

Comment:
This will panic if the node restarts since "Store" overwrites the previous entry which is disabled for the models in goshimmer.

Maybe better use:

```suggestion
	b.tangle.Storage.MessageMetadata(EmptyMessageID, func() *MessageMetadata {
		return genesisMetadata
	}).Release()
```